### ISSUE TITLE
When run with sudo ... the resulting AppImage has root as its owner

### ISSUE THREAD

**Issue by probonopd** [AppImage/AppImageUpdate#118]
When AppImageUpdate is run with sudo (as I need to do in order to update things in `/isodevice` (the device Ubuntu mounts the medium on which the Live ISO resides to), then the resulting "new" AppImage has root as its owner, which is not what we want.

Hence, we should run `sudo -R $USER $New.AppImage` at the end.

**Comment by TheAssassin** [comment]
`$USER == root` if `sudo` is used.

**Comment by probonopd** [comment]
I am trying on Ubuntu.

**Comment by TheAssassin** [comment]
Please research a bit more on the topic. IIRC there's a variable `sudo` exports that shows the original user. But this seems quite implicit...

Did you mean to write `chown -R ...`? (That `-R` is by the way not necessary)

**Comment by probonopd** [comment]
Sure, I meant `sudo chown`, corrected above.

**Comment by TheAssassin** [comment]
Shouldn't we just add a flag to copy the old owner and their permissions? Like `-p` or so? Changing owner would only work properly when you call it with `sudo`, and it's less implicit than using `$SUDO_USER`.

Or maybe we should just do that by default? Not a fan, but perhaps this is desirable for some people.

**Comment by probonopd** [comment]
> Copy the old owner and their permissions
> Or maybe we should just do that by default? 

Yes, absolutely. Copy over owner and permissions from the old to the new file.

**Comment by TheAssassin** [comment]
But only if a flag is passed, at least for `zsync2`, for AppImageUpdate I'd consider adding a pre-checked checkbox to create an opt-out.

**Comment by probonopd** [comment]
Yes :+1:

**Comment by shoogle** [comment]
> @TheAssassin 
> `$USER == root` if `sudo` is used.

> @probonopd 
> No:
> 
> ```
> me@host:~$ sudo echo $USER
> me
> ```

Variable substitution is performed by the shell (which is not root) before arguments are passed to the `sudo` command. Try this instead:

```bash
me@host:~$ sudo bash -c 'echo $USER'
root
```

**Comment by TheAssassin** [comment]
Note the `'` single quotes, which escape the contents.

**Comment by shoogle** [comment]
Indeed, it would not work with `"` double quotes. (It would return "me" instead of "root".)
